### ISSUE TITLE
Ongoing issues with custom sound fonts on NXTs

### ISSUE THREAD

**Issue by jramboz** [jramboz/tintalle#15]
See jramboz/py2saber#2

**Comment by jramboz** [comment]
Seems like there's definitely something on the firmware side that needs to be updated to fully address this issue. I know LdL is working on it.

**Comment by jramboz** [comment]
Based on a tip from Mitriax on Discord, it seems that custom sound fonts work as long as the lengths of sound files are less than:

- Hum: 20 seconds
- Smoothswings: 6 seconds
- Poweron: 3 seconds

I'm doing some more testing, but so far at least a few sound fonts that formerly locked up the NXT appear to be working stably.

**Comment by mrseeker** [comment]
I have a power-on sound that is 10sec long, and that worked?

**Comment by jramboz** [comment]
Yeah. These numbers definitely are not perfect. I have some sound fonts that still cause problems even with these values, and others that are longer and work fine. So far, I haven't been able to figure out a pattern. But these settings seem to work for the majority of sound fonts.

**Comment by mrseeker** [comment]
So, from what I have witnessed, the power-on and power-off sound fonts really depend on the fact that they are interposed on each other. So, the hum and power-on will play at the same time, however if the power-on is longer than the hum, the power-on will play first, followed by the hum being added to the power-up, making it so that they both finish at the same time. The power-off works in a similar way, but the light will stop at the same time the power-off starts playing. I also found out that I can have a power-off for as long as I want, as long as it ends with a sound. You need to export your sounds with Mono PCM 16Bit 44100hz (use audacity export).

**Comment by jramboz** [comment]
Thread necromancy!

I think there may be some issues involving the Clash effects as well. I've noticed that with some sound fonts, a Clash will start playing on a loop and the saber will lock up. This seems to happen after a fairly regular amount of time (a minute or two, maybe?), but that will need more testing. I can further experiment with Clash timing to see if that has any effect.

**Comment by jramboz** [comment]
The Clashes were a false lead. However, I've determined that the issue (or at least, _an_ issue) is with the Hum.

I've isolated two Hum files, one from a working sound font, and one from a sound font that causes lockup. I uploaded ONLY the Hums (plus the Beep) to the Anima. I get consistent results every time: the "good" file repeats indefinitely without issue; the "bad" one causes a lockup and silence after exactly two repeats.

**Comment by mrseeker** [comment]
```
I've isolated two Hum files, one from a working sound font, and one from a sound font that causes lockup. I uploaded ONLY the Hums (plus the Beep) to the Anima. I get consistent results every time: the "good" file repeats indefinitely without issue; the "bad" one causes a lockup and silence after exactly two repeats.
```
Any differences in the bitrate, sample rate etc?

**Comment by jramboz** [comment]
No, nothing obvious. I've been working with Claude to try to isolate the factor that's causing the lockup. So far, no luck. I'll continue with it, but it's a bit slow since I'm on the free plan and limited in how many messages I get. 😄

**Comment by mrseeker** [comment]
Here's the thing: Unless you can differentiate between the wav files, you are essentially testing against a black box. I can help you on Discord (I use the same username), since the wav files did not seem to be uploaded here.

**Comment by jramboz** [comment]
I'll send them to you on Discord. I'm hesitant to post them here since they're from commercial sound fonts. But I don't mind sending them for testing purposes. 

Also, Black Box testing is essentially what I've been doing: altering various properties, creating test files, and uploading them to an Anima to see what happens.
